pro/scripts/generate_sitemap.py

```diff
@@ -0,0 +1,214 @@
+
+    to_create.sort(key=lambda x: int(x["order"].split('-')[depth]))
+
+    previous = 0
+    for elem in to_create:
+        order_parts = elem["order"].split('-')
+        o = int(order_parts[depth])
+        res = get_nested_node(root, order_parts, depth) if depth > 0 else root
+        if previous + 1 < o:
+            insert_missing_nodes(res, o - previous - 1)
+
+        if elem["parent"]:
+            parent_order_parts = order_parts[:-1]
+            parent_node = get_nested_node(root, parent_order_parts, depth - 1)
+            parent_node[int(parent_order_parts[-1]) - 1]["title"] = elem["parent"]
+
+        res.append(elem)
+        previous = o
+
+    insert_node(root, nodes, depth + 1)
+
+def transform_routes(data):
+    """
+    Transforme les routes en une structure imbriquée
+    """
+
+    res = []
+    nodes = []
+
+    # Build initial nodes
+    for item in data:
+        nodes.append({
+            "title": item['title'],
+            "path": item['path'],
+            "parent": item['parent'],   
+            "order": item['order'],         
+            "depth": len(item['order'].split('-')) - 1,
+            "children": [],
+        })
+
+    insert_node(res, nodes)
+
+    return res
+
+
+def generate_sitemap_component(routes):
+    """
+    Génère le composant Sitemap à partir des routes
+    """
+    def generate_list(routes, depth=0):
+        indent = '        ' * depth
+        list_items = ""
+        for route in routes:
+            path = route.get('path', '')
+            title = route.get('title', '')
+            children = route.get('children', [])
+            if path:
+                list_items += f"""<li className={{styles['sitemap-list-item']}}>
+{indent}  <Link to={{`{path.replace(':offererId', '${selectedOffererId}')}`}} className={{styles['sitemap-link']}}>
+{indent}    {title}
+{indent}  </Link>"""
+                if children:
+                    list_items += f"""
+{indent}  <ul className={{styles['sitemap-sub-list']}}>
+{generate_list(children, depth + 1)}{indent}  </ul>"""
+                list_items += f"""
+{indent}</li>"""
+            else:
+                list_items += f"""
+<li className={{styles['sitemap-list-item']}}>
+{indent}  <span className={{styles['sitemap-list-title']}}>{title}</span>"""
+                if children:
+                    list_items += f"""
+{indent}  <ul className={{styles['sitemap-sub-list']}}>
+{generate_list(children, depth + 1)}
+{indent}  </ul>"""
+                list_items += f"""
+{indent}</li>"""
+        return list_items
+
+    component = f"""//
+// Generated by generate_sitemap.py, do not modify. Launch 'yarn run generate:sitemap' to update
+//
+
+import React from 'react'
+import {{ useSelector }} from 'react-redux'
+import {{ Link }} from 'react-router-dom'
+
+import {{ AppLayout }} from 'app/AppLayout'
+import {{ selectCurrentOffererId }} from 'store/user/selectors'
+
+import styles from './Sitemap.module.scss'
+
+export const Sitemap = () => {{
```

Comment:
Et avoir ce template ici va être assez relou à modifier/itérer quand on va vouloir le changer (par ex les imports, les composants qui sont utilisés dedans etc)